Add `transformRequest` prop to map components.

colllin opened the pull request:
- [x] Add `transformRequest` to props and map constructor
- [x] Document `transformRequest` prop
- [x] Test `transformRequest` prop
- [x] Upgrade `sinon` to fix `test-browser` script
- [ ] Fix map `onLoad` tests

The mapboxgl `transformRequest` map option is necessary when you need to modify the request headers, for example when your data sources require an `Authorization` header.

Example from our app:

```
    transformRequest = (url, resourceType) => {
        if (resourceType == 'Tile' && url.startsWith('https://example.authenticated.com')) {
            return {
                url: url,
                headers: {
                    Authorization: 'Bearer $token'.replace('$token', localStorage.getItem('authToken'))
                }
            }
        }
    }

    render() {
        return (
            <ReactMapGL ... transformRequest={this.transformRequest} />
        )
    }
```

Unfortunately, there's no way to set the map's `transformRequest` option outside of the constructor.

It might be worth considering passing-through all unknown props to the `mapboxgl.map()` constructor, rather than dealing with them individually like this, but I didn't want to make that sweeping decision or tackle that problem right now.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/react-map-gl?pullRequest=426) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/react-map-gl?pullRequest=426) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/react-map-gl?pullRequest=426) it.</sub>

colllin commented:
Question on the tests:

I fixed the browser test for the map `onLoad` prop.  The problem is, both the test was broken **and** *the test was failing* when I fixed it.  Here's the test output:

```
# InteractiveMap#call onLoad when provided
ok 121 InteractiveMap rendered
not ok 122 onLoad does not expose the map object.
  ---
    operator: equal
    expected: 0
    actual:   1
    stack: |-
      assert@http://localhost:8080/test-browser-bundle.js:64458:63
      bound@http://localhost:8080/test-browser-bundle.js:64310:37
      strictEquals@http://localhost:8080/test-browser-bundle.js:64593:17
      bound@http://localhost:8080/test-browser-bundle.js:64310:37
      onLoad@http://localhost:8080/test-browser-bundle.js:68632:9
      fire@http://localhost:8080/test-browser-bundle.js:45152:973
      _render@http://localhost:8080/test-browser-bundle.js:45126:14961
      [native code]
  ...
```

So, if I change the `onLoad` callback to no longer receive the `event` argument, thus fixing the test by preventing access to the mapbox instance within the callback... this could be a potentially breaking change if anyone has been relying on having access to that event argument.  I think the original design intention makes sense (to avoid "leaking" the mapbox instance here), but I wanted to confirm before making a potentially breaking change/fix.

As-is, this `onLoad` test is the only failing browser test.

ibgreen commented on src/mapbox/mapbox.js:
```diff
@@ -66,6 +67,7 @@ const defaultProps = {
   onLoad: noop,
   onError: noop,
   reuseMaps: false,
+  transformRequest: undefined,
```
Use `null` instead of `undefined`? I find explicit usage of `undefined` to be tricky as it so easily can be the result of an unintentional coding error.

colllin replied:
Good catch.  Changed default to `null` and now I only pass it to the constructor if present.  Thanks for the review!
